deploy/runtools/run_farm.py

```diff
@@ -1,361 +1,397 @@
 
-        for index in range(len(self.f1_2s)):
-            self.f1_2s[index].assign_boto3_instance_object(MockBoto3Instance())
+                assert (len(items) == 1), f"dict type 'runhost' items map a single EC2 instance name to a number. Not {pprint.pformat(runhost)}"
 
-        for index in range(len(self.m4_16s)):
-            self.m4_16s[index].assign_boto3_instance_object(MockBoto3Instance())
+                inst_handle, num_insts = next(iter(items))
+
+                if inst_handle in self.SIM_HOST_HANDLE_TO_MAX_FPGA_SLOTS:
+                    num_sim_slots = self.SIM_HOST_HANDLE_TO_MAX_FPGA_SLOTS[inst_handle]
+                    insts: List[Tuple[Inst, Optional[Union[EC2InstanceResource, MockBoto3Instance]]]] = []
+                    for n in range(num_insts):
+                        insts.append((Inst(num_sim_slots, EC2InstanceDeployManager, self.default_simulation_dir), None))
+                    self.run_farm_hosts_dict[inst_handle] = insts
+                    self.mapper_consumed[inst_handle] = 0
+                else:
+                    rootLogger.critical(f"WARNING: Skipping {inst_handle} since it is not supported. Use {self.SIM_HOST_HANDLE_FOR_SWITCH_ONLY_SIM}.")
+            else:
+                raise Exception(f"Unknown runhost handle of {runhost}")
+
+    def get_smallest_sim_host_handle(self, num_sims: int) -> str:
+        for max_simcount, sim_host_handle in self.SORTED_SIM_HOST_HANDLE_TO_MAX_FPGA_SLOTS:
+            if max_simcount < num_sims:
+                # instance doesn't support enough sims
+                continue
+            num_consumed = self.mapper_consumed[sim_host_handle]
+            num_allocated = len(self.run_farm_hosts_dict[sim_host_handle])
+            if num_consumed >= num_allocated:
+                # instance supports enough sims but none are available
+                continue
+            return sim_host_handle
+
+        rootLogger.critical(f"ERROR: No instances are available to satisfy the request for an instance with support for {num_sims} simulation slots. Add more instances in your run farm configuration (e.g., config_run_farm.ini).")
+        raise Exception
+
+    def allocate_sim_host(self, sim_host_handle: str) -> Inst:
+        inst_tup = self.run_farm_hosts_dict[sim_host_handle][self.mapper_consumed[sim_host_handle]]
+        inst_ret = inst_tup[0]
+        self.mapper_consumed[sim_host_handle] += 1
+        return inst_ret
+
+    def get_default_switch_host_handle(self) -> str:
+        return self.SIM_HOST_HANDLE_FOR_SWITCH_ONLY_SIM
+
+    def bind_mock_instances_to_objects(self) -> None:
+        """ Only used for testing. Bind mock Boto3 instances to objects. """
+        for inst_handle, inst_list in self.run_farm_hosts_dict.items():
+            for idx, run_farm_host_tup in enumerate(inst_list):
+                boto_obj = MockBoto3Instance()
+                inst = run_farm_host_tup[0]
+                inst.set_host("centos@" + boto_obj.private_ip_address)
```

Comment:
I added centos everywhere here to make it clear what was running on centos (vs. not). I can remove this in the places that I added it so that it matches old functionality.

Reply:
I think it's probably a good idea to revert it.